packages/react/src/Timeline/Timeline.tsx

```diff
@@ -1,76 +1,158 @@
+  'div',
+  styled.div<StyledTimelineProps>`
+    display: flex;
+    flex-direction: column;
+    ${props =>
+      props.clipSidebar &&
+      css`
+        .Timeline-Item:first-child {
+          padding-top: 0;
+        }
+
+        .Timeline-Item:last-child {
+          padding-bottom: 0;
+        }
+      `}
+
+    ${sx};
+  `,
+)
+
+export type TimelineProps = StyledTimelineProps & HTMLProps<HTMLDivElement>
+
+const Timeline = React.forwardRef<HTMLElement, TimelineProps>(function Timeline(
+  {clipSidebar, className, ...props},
+  forwardRef,
+) {
+  const enabled = useFeatureFlag(CSS_MODULES_FEATURE_FLAG)
+  if (enabled) {
+    return (
+      <ToggleTimeline
+        {...props}
+        className={clsx(className, classes.Timeline)}
+        ref={forwardRef}
+        data-clip-sidebar={clipSidebar ? '' : undefined}
+      />
+    )
+  }
+
+  return <ToggleTimeline {...props} className={className} ref={forwardRef} clipSidebar={clipSidebar} />
+})
+
+Timeline.displayName = 'Timeline'
+
+type StyledTimelineItemProps = {condensed?: boolean; className?: string} & SxProp
+
+const ToggleTimelineItem = toggleStyledComponent(
+  CSS_MODULES_FEATURE_FLAG,
+  'div',
+  styled.div.attrs<StyledTimelineItemProps>(props => ({
+    className: clsx('Timeline-Item', props.className),
+  }))<StyledTimelineItemProps>`
+    display: flex;
+    position: relative;
+    padding: ${get('space.3')} 0;
+    margin-left: ${get('space.3')};
+
+    &::before {
+      position: absolute;
+      top: 0;
+      bottom: 0;
+      left: 0;
+      display: block;
+      width: 2px;
+      content: '';
+      background-color: ${get('colors.border.muted')};
+    }
+
+    ${props =>
+      props.condensed &&
+      css`
+        padding-top: ${get('space.1')};
         padding-bottom: 0;
-      }
-    `}
-
-  ${sx};
-`
-
-type StyledTimelineItemProps = {condensed?: boolean} & SxProp
-
-const TimelineItem = styled.div.attrs<StyledTimelineItemProps>(props => ({
-  className: clsx('Timeline-Item', props.className),
-}))<StyledTimelineItemProps>`
-  display: flex;
-  position: relative;
-  padding: ${get('space.3')} 0;
-  margin-left: ${get('space.3')};
-
-  &::before {
-    position: absolute;
-    top: 0;
-    bottom: 0;
-    left: 0;
-    display: block;
-    width: 2px;
-    content: '';
-    background-color: ${get('colors.border.muted')};
+        &:last-child {
+          padding-bottom: ${get('space.3')};
+        }
+
+        .TimelineItem-Badge {
+          height: 16px;
+          margin-top: ${get('space.2')};
+          margin-bottom: ${get('space.2')};
+          color: ${get('colors.fg.muted')};
+          background-color: ${get('colors.canvas.default')};
+          border: 0;
+        }
+      `}
+
+    ${sx};
+  `,
+)
+
+/**
+ * @deprecated Use the `TimelineItemProps` type instead
+ */
+export type TimelineItemsProps = StyledTimelineItemProps & HTMLProps<HTMLDivElement>
+
+export type TimelineItemProps = StyledTimelineItemProps & HTMLProps<HTMLDivElement>
```

Comment:
It seems to me the "correct" prop name would be `TimelineItemProps` singular, so I'm proposing deprecating `TimelineItemsProps` in favor of `TimelineItemProps`. Would love to hear thoughts on this one